proto/api.go

```diff
@@ -86,6 +86,12 @@ type Request interface {
 	flags() int
 }
 
+// RequestWithUser is an interface for RPC requests that have a "requested user".
+type RequestWithUser interface {
```

Comment:
`UserRequest`? Note that there is no need to define this interface here. You could define it in rpc/codec right next to its use. (And name it `userRequest` or something like that).